src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -168,8 +168,15 @@ bool TargetedMovementGeneratorMedium<T, D>::DoUpdate(T* owner, uint32 time_diff)
     if (i_recheckDistance.Passed())
     {
         i_recheckDistance.Reset(100);
+
         //More distance let have better performance, less distance let have more sensitive reaction at target move.
-        float allowed_dist = owner->GetCombatReach() + sWorld->getRate(RATE_TARGET_POS_RECALCULATION_RANGE);
+        float allowed_dist = 0.0f;
+
+        if ((owner->GetTypeId() == TYPEID_UNIT && owner->IsPet()) && (owner->GetCharmerOrOwnerGUID() == i_target->GetGUID()))
```

Comment:
so does 6.x, you must have used grep or viewed it in some other form completely out of context - GetHigh is a method inside ObjectGuid, Unit::IsPet is `m_unitTypeMask & UNIT_MASK_PET`